OtripleS.Web.Api.Tests.Unit/Services/SemesterCourseServiceTests/SemesterCourseServiceTests.Exceptions.Delete.cs

```diff
@@ -0,0 +1,165 @@
+
+            this.loggingBrokerMock.Verify(broker =>
+                    broker.LogCritical(It.Is(SameExceptionAs(expectedSemesterCourseDependencyException))),
+                Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                    broker.SelectSemesterCourseByIdAsync(inputSemesterCourseId),
+                Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnDeleteWhenDbExceptionOccursAndLogItAsync()
+        {
+            // given
+            Guid randomSemesterCourseId = Guid.NewGuid();
+            Guid inputSemesterCourseId = randomSemesterCourseId;
+            var databaseUpdateException = new DbUpdateException();
+
+            var expectedSemesterCourseDependencyException =
+                new SemesterCourseDependencyException(databaseUpdateException);
+
+            this.storageBrokerMock.Setup(broker =>
+                    broker.SelectSemesterCourseByIdAsync(inputSemesterCourseId))
+                .ThrowsAsync(databaseUpdateException);
+
+            // when
+            ValueTask<SemesterCourse> deleteSemesterCourseTask =
+                this.semesterCourseService.DeleteSemesterCourseAsync(inputSemesterCourseId);
+
+            // then
+            await Assert.ThrowsAsync<SemesterCourseDependencyException>(() =>
+                deleteSemesterCourseTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                    broker.LogError(It.Is(SameExceptionAs(expectedSemesterCourseDependencyException))),
+                Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                    broker.SelectSemesterCourseByIdAsync(inputSemesterCourseId),
+                Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnDeleteWhenDbUpdateConcurrencyExceptionOccursAndLogItAsync()
+        {
+            // given
+            Guid randomSemesterCourseId = Guid.NewGuid();
+            Guid inputSemesterCourseId = randomSemesterCourseId;
+            var databaseUpdateConcurrencyException = new DbUpdateConcurrencyException();
+
+            var lockedSemesterCourseException = new LockedSemesterCourseException(databaseUpdateConcurrencyException);
+
+            var expectedSemesterCourseException =
+                new SemesterCourseDependencyException(lockedSemesterCourseException);
+
+            this.storageBrokerMock.Setup(broker =>
+                    broker.SelectSemesterCourseByIdAsync(inputSemesterCourseId))
+                .ThrowsAsync(databaseUpdateConcurrencyException);
```

Comment:
Fix this

Reply:
I guess the exception should be thrown when I try to call *DeleteSemesterCourseAsync* right?

Reply:
I meant push the tabs - the code is doing the right thing.

Reply:
😅 ok i see, I have tried to fix the tabs issues. i will review them again

Reply:
Thanks, I have fixed the tab alignment, Please check if its ok now.